File: human_chat.py
t = int(input())

while t > 0:
    n = int(input())
    a = list(map(int, input().split()))

    ans = [0] * n
    st = []

    for i in range(n - 1, -1, -1):
        while st and a[st[-1]] <= a[i]:
            st.pop()

        if not st:
            ans[i] = n - i - 1
        else:
            ans[i] = st[-1] - i

        st.append(i)

    print(*ans)

    t -= 1

File: human_chat.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;

        vector<int> a(n);
        vector<int> ans(n);

        for (int i = 0; i < n; i++) {
            cin >> a[i];
        }

        stack<int> st;

        for (int i = n - 1; i >= 0; i--) {
            while (!st.empty() && a[st.top()] <= a[i]) {
                st.pop();
            }

            if (st.empty()) {
                ans[i] = n - i - 1;
            } else {
                ans[i] = st.top() - i;
            }

            st.push(i);
        }

        for (int i = 0; i < n; i++) {
            cout << ans[i] << " ";
        }

        cout << endl;
    }

    return 0;
}